aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -170,16 +183,26 @@ SparseTensor& div_out_sparse_zerodim(SparseTensor& r, const SparseTensor& t, con
   AT_ASSERT(t.is_sparse());
 
   if (is_same_tensor(r, t)) {
+    // Can't divide an uncoalesced integral tensor accurately. e.g. for a sparse int tensor with value 6
+    // represented as values=[3, 3], integer division by 2 would give values=[1, 1] => 2 instead
+    // of 6 / 2 => 3
+    if (!r.is_coalesced() && isIntegralType(r.scalar_type(), /*includeBool=*/true)) {
+      coalesce_(r);
+    }
     r._values().div_(value);
   } else {
-    r.resize_as_(t);
+    Tensor t_tmp = t;
+    if (!t.is_coalesced() && isIntegralType(r.scalar_type(), /*includeBool=*/true)) {
+      t_tmp = t.coalesce();
```

Comment:
We can actually do this in-place, by first copying `t` to `r`, doing in-place coalesce on `r`, and finally performing the division. However, based on the implementation of `coalesce_` this doesn't save any extra time or memory (because our in-place coalesce actually does it out of place...) so I don't think this matters